src/compose/service.coffee

```diff
@@ -0,0 +1,343 @@
+		@environment = newEnv
+		return @environment
+
+	extendLabels: (imageInfo) =>
+		@labels = _.clone(@labels)
+		_.defaults(@labels, imageInfo?.Config?.Labels ? {})
+		@labels['io.resin.supervised'] = 'true'
+		@labels['io.resin.app_id'] = @appId.toString()
+		@labels['io.resin.service_id'] = @serviceId.toString()
+		@labels['io.resin.service_name'] = @serviceName
+		@labels['io.resin.image_id'] = @imageId.toString()
+		@labels['io.resin.commit'] = @commit
+		return @labels
+
+	extendAndSanitiseExposedPorts: (imageInfo) =>
+		@expose = _.clone(@expose)
+		@expose = _.map(@expose, String)
+		if imageInfo?.Config?.ExposedPorts?
+			_.forEach imageInfo.Config.ExposedPorts, (v, k) =>
+				port = k.match(/^([0-9]*)\/tcp$/)?[1]
+				if port? and !_.find(@expose, port)
+					@expose.push(port)
+
+		return @expose
+
+	extendAndSanitiseVolumes: (imageInfo) =>
+		volumes = []
+		_.forEach @volumes, (vol) ->
+			isBind = /:/.test(vol)
+			if isBind
+				bindSource = vol.split(':')[0]
+				if !path.isAbsolute(bindSource)
+					volumes.push(vol)
+				else
+					console.log("Ignoring invalid bind mount #{vol}")
+			else
+				volumes.push(vol)
+		volumes = volumes.concat(@defaultBinds())
+		volumes = _.union(_.keys(imageInfo?.Config?.Volumes), volumes)
+		@volumes = volumes
+		return @volumes
+
+	getNamedVolumes: =>
+		defaults = @defaultBinds()
+		validVolumes = _.map @volumes, (vol) ->
+			return null if _.includes(defaults, vol)
+			return null if !/:/.test(vol)
+			bindSource = vol.split(':')[0]
+			if !path.isAbsolute(bindSource)
+				return bindSource
+			else
+				return null
+		return _.filter(validVolumes, (v) -> !_.isNull(v))
+
+	lockPath: =>
+		return updateLock.lockPath(@appId)
+
+	killmePath: =>
+		return killmePath(@appId, @serviceName)
+
+	killmeFullPathOnHost: =>
+		return "#{constants.rootMountPoint}#{@killmePath()}/resin-kill-me"
+
+	defaultBinds: ->
+		return defaultBinds(@appId, @serviceName)
+
+	@fromContainer: (container) ->
+		if container.State.Running
+			status = 'Running'
+		else if container.State.Status == 'created'
+			status = 'Installed'
+		else
+			status = 'Stopped'
+
+		boundContainerPorts = []
+		ports = []
+		expose = []
+		_.forEach container.HostConfig.PortBindings, (conf, port) ->
+			containerPort = port.match(/^([0-9]*)\/tcp$/)?[1]
+			if containerPort?
+				boundContainerPorts.push(containerPort)
+				hostPort = conf[0]?.HostPort
+				if !_.isEmpty(hostPort)
+					ports.push("#{hostPort}:#{containerPort}")
+				else
+					ports.push(containerPort)
+		_.forEach container.Config.ExposedPorts, (conf, port) ->
+			containerPort = port.match(/^([0-9]*)\/tcp$/)?[1]
+			if containerPort? and !_.includes(boundContainerPorts, containerPort)
+				expose.push(containerPort)
+
+		appId = container.Config.Labels['io.resin.app_id']
+		serviceId = container.Config.Labels['io.resin.service_id']
+		serviceName = container.Config.Labels['io.resin.service_name']
+		releaseId = container.Name.match(/.*_(\d+)$/)?[1]
+		service = {
```

Comment:
This object has inconsistent casing. Some are camelCase some snake_case

Reply:
This is because some fields come from the compose yaml snake_cased, and we use camelCase for everything else... I guess the best way to address this is to have the builder change everything to camelCase in its compose.yml parser - so I'll defer this until we have a way forward on that unified parser and how it works, sound ok?

Reply:
Fixed with help from `_.camelCase`